### PR TITLE
Feature/check ordering js

### PR DESCRIPTION
**Pull request by johnhelmuth**
Added script `scripts/check-csv-order.js` to do the check in a system agnostic way, so we don't have to worry about having the same versions of unix utilities installed to do the check locally.

Added dev dependency on node js module [`csv-parse`](https://github.com/adaltas/node-csv/tree/master/packages/csv-parse/) for the script to read and parse the CSV file so that the Publisher Name and Product title can be compared independently.

Hooked that script into `package.json` as the package test script.

Modified `.github/workflows/check-csv-ordering.yaml` to use nodejs and the `npm test` script for the check.

Updated `readme.md` with instructions on running the check locally.

Added `.gitattributes` file to automatically identify text files and to enable line ending normalization.

Normalized CSV file line endings in the repository.

Let me know if you run into issues with line endings on this file.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -16,6 +16,43 @@ The submission will be reviewed and added to the Fate SRD. This usually happens
 
 If you are comfortable with editing files and git/Github, you may [create a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request). It will be reviewed and added to the site.
 
+There is a Github check configured to ensure the CSV file is in the expected order. This check is implemented a small
+Javascript script that runs under [Nodejs](https://nodejs.org/) version 20.x. (The code ill probably run under older versions, too.)
```
Comment (amazingrando) [inline readme.md:20]: Change "ill" to "will" in "...The code ill probably..." to fix a typo.

Comment (amazingrando) [inline readme.md:20]: Do we want to formally declare node 20 with an .npmrc file in the repo?

**Review by amazingrando — CHANGES_REQUESTED** [review]
I left a comment and question. Here is a request—add a .gitignore file and block `node_modules/` from the repo.

After that, this is good to go!

File: readme.md (hunk 1 above)
Comment (johnhelmuth) [reply readme.md:20]: Another way to do that is via the "[engines](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines)" key in package.json. TBH, I thought I had done that already.

Do you have a preference for which technique to use?  I don't know if there is a functional difference; I think putting it in the package.json file makes it more obvious, developers will more likely read package.json than a hidden file in the repo.

Comment (johnhelmuth) [reply readme.md]: I tested the code with node 18 and node 19, both threw an error on the usage of `Array.prototype.toSorted()`, so I have added in an `"engines": { "node": "20.x" }` stanza to package.json, which causes node 18 and node 19 fail the npm install (or npm ci) steps.

**Pull request merged**